core/src/operations/decorators/debug.rs

```diff
@@ -0,0 +1,25 @@
+use core::fmt;
+
+// DEBUG OPTIONS
+// ================================================================================================
+
+/// Options of the `Debug` decorator.
+///
+/// These options define the debug info which gets printed out when the Debug decorator is
+/// executed.
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum DebugOptions {
+    /// Print out the entire contents of the stack for the current execution context.
+    StackAll,
+    /// Prints out the top n items of the stack for the current context.
+    StackTop(u8),
+}
+
+impl fmt::Display for DebugOptions {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::StackAll => write!(f, "stack.all"),
```

Comment:
I don't think this is consistent with the way that the syntax is written, I would expect just "stack" without the postfix ".all", however I do think it adds valuable contextual information so maybe it's fine to keep.

Reply:
Actually, this was inconsistent with how the parser works (which expects just `debug.stack`. So, I removed the "all" part.